src/main/java/duke/task/TaskList.java

```diff
@@ -0,0 +1,134 @@
+package duke.task;
+
+import java.util.ArrayList;
+
+import duke.core.ChatbotException;
+import duke.storage.Storage;
+
+/**
+ * Represents a TaskList that stores all tasks specified by user.
+ */
+public class TaskList {
+    public static int tasksCount = 0;
+
+    private ArrayList<Task> tasks;
+
+    /**
+     * Initialises a TaskList with an array to store the tasks.
+     */
+    public TaskList() {
+        this.tasks = new ArrayList<>();
+        Storage.ensureFileExists();
+        Storage.loadTasks(tasks);
+    }
+
+    /**
+     * Adds user-specified task into the tasklist
+     *
+     * @param input Task to be added into the tasklist.
+     * @return Boolean if it's a valid task.
+     */
+    public boolean addTask(String input) {
+        Task task = convertTask(input);
+        if (task != null) {
+            tasks.add(task);
+            tasksCount++;
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Deletes user-specified task from the tasklist
+     *
+     * @param num Number associated with the task.
+     */
+    public void deleteTask(int num) {
+        tasks.remove(num);
+        tasksCount--;
+    }
+
+    /**
+     * Creates Task while ensuring user input is valid
+     *
+     * @param input User-specified task.
+     * @return Task if valid user input, else return null.
+     */
+    public Task convertTask(String input) {
+
+        if (input.trim().isEmpty()) {
+            ChatbotException.getError(ChatbotException.ErrorType.UNKNOWN_COMMAND);
+            return null;
+        }
+
+        String[] inputs = input.split("\\s+", 2);
+        if (inputs.length < 2 || inputs[1].trim().isEmpty()) {
+            catchInputError(inputs[0]);
+            return null;
+        }
+
+        return Storage.createTask(inputs[0], inputs[1]);
+    }
+
+    /**
+     * Lists all tasks stored in tasklist
+     */
+    public void listTask() {
+        for (int i = 0; i < tasksCount; i++) {
+            System.out.println((i + 1) + ". " + tasks.get(i).getDescription());
+        }
+    }
+
+    /**
+     * Updates status of the user-specified task
+     *
+     * @param num Number associated with the task.
+     * @param status Status if the task is done or not done.
+     */
+    public void markTask(int num, boolean status) {
+        tasks.get(num).mark(status);
+    }
+
+    /**
+     * Finds all tasks associated with the given keyword
+     *
+     * @param keyword User-specified keyword
+     */
+    public void findTasks(String keyword) {
+        ArrayList<Task> foundTasks = new ArrayList<>();
+
+        for (Task task : tasks) {
+            if (task.getInitialDesc().toLowerCase().contains(keyword.toLowerCase())) {
+                foundTasks.add(task);
+            }
+        }
+
+        for (int i = 0; i < foundTasks.size(); i++) {
+            System.out.println((i + 1) + ". " + foundTasks.get(i).getDescription());
+        }
+    }
```

Comment:
Consider returning an `ArrayList<Task>`, or even better, a `TaskList` object, then using the `Ui` class to print the output. i.e.
`return foundTasks` 
and in the Ui class:
```
public void printList(ArrayList<Task> tasks) {
    for (int i = 0; i < foundTasks.size(); i++) {
            System.out.println((i + 1) + ". " + foundTasks.get(i).getDescription());
}